htdocs/takepos/index.php

```diff
@@ -824,7 +830,7 @@ function WeighingScale(){
 
 <body class="bodytakepos" style="overflow: hidden;">
 <?php
-$keyCodeForEnter = getDolGlobalInt('CASHDESK_READER_KEYCODE_FOR_ENTER'.$_SESSION['takeposterminal']) > 0 ? getDolGlobalInt('CASHDESK_READER_KEYCODE_FOR_ENTER'.$_SESSION['takeposterminal']) : '';
+$keyCodeForEnter = getDolGlobalInt('CASHDESK_READER_KEYCODE_FOR_ENTER'. (isset($_SESSION["takeposterminal"])) ? $_SESSION["takeposterminal"] : '') > 0 ? getDolGlobalInt('CASHDESK_READER_KEYCODE_FOR_ENTER'. (isset($_SESSION["takeposterminal"])) ? $_SESSION["takeposterminal"] : '') : '';
```

Comment:
Something wrong here
getDolGlobalInt('CASHDESK_READER_KEYCODE_FOR_ENTER'. (isset($_SESSION["takeposterminal"])) ? $_SESSION["takeposterminal"] : '') : '';

What is done in priority ?  The concat "." then the "?" or the "?" and after the ".".  Use parenthesis to decide which order you want.